compat/xaccept.cc

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+#include "compat/xaccept.h"
+
+#if _SQUID_WINDOWS_ || _SQUID_MINGW_
+#if HAVE_WINDOWS_H
+#include <windows.h>
+#endif
+#if HAVE_WKINSOCK2_H
+#include <winsock2.h>
+#endif
+
+int
+xaccept(int s, struct sockaddr *a, socklen_t *l)
+{
+    SOCKET result;
+    if ((result = accept(_get_osfhandle(s), a, l)) == INVALID_SOCKET)
+    {
+        if (WSAEMFILE == (errno = WSAGetLastError()))
+            errno = EMFILE;
+        return -1;
+    }
+    else
+        return _open_osfhandle(result, 0);
```

Comment:
AFAICT, _open_osfhandle() [requires](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/open-osfhandle?view=msvc-170) _close(). Do we call _close() for the returned file handler on MinGW? FWIW, I cannot find the corresponding MinGW code wrapping close(), but there is one for WIndows in os/mswindows.h. If MinGW build would fail without a close() wrapper, then this PR may go in first. Otherwise, we would have to wrap close() first.

This comment is only asking a question.

Reply:
I believe MinGW will fail to build `close()` passing it a `SOCKET` instead of `int`, same as for `accept()` and other socket calls.

Reply:
Yes, that much is clear. Since regular Squid code does _not_ pass `SOCKET` to close() -- it passes `int` that this accept() wrapper returns to it -- the question is whether MinGW build currently fails due to, among other problems, a missing close() declaration. If it fails for that reason (among others), then this PR can go in first.